webui/react/src/pages/ExperimentDetails/ExperimentVisualization/HpHeatMaps.tsx

```diff
@@ -247,19 +247,25 @@ const HpHeatMaps: React.FC<Props> = ({
             .filter(hParam => fullHParams.includes(hParam))
             .sort((a, b) => a.localeCompare(b, 'en'));
 
+          /**
+           * TODO: filtering NaN, +/- Infinity for now, but handle it later with
+           * dynamic min/max ranges via uPlot.Scales.
+           */
+          const trialMetric = Number.isFinite(trial.metric) ? trial.metric : null;
+
           trialIds.push(trialId);
           hpMetricMap[trialId] = hpMetricMap[trialId] || {};
           hpValueMap[trialId] = hpValueMap[trialId] || {};
           trialHParams.forEach(hParam1 => {
             hpValueMap[trialId][hParam1] = flatHParams[hParam1];
             trialHParams.forEach(hParam2 => {
               const key = generateHpKey(hParam1, hParam2);
-              hpMetricMap[trialId][key] = trial.metric;
+              hpMetricMap[trialId][key] = trialMetric;
             });
           });
 
-          if (trial.metric < metricRange[0]) metricRange[0] = trial.metric;
-          if (trial.metric > metricRange[1]) metricRange[1] = trial.metric;
+          if (trialMetric !== null && trialMetric < metricRange[0]) metricRange[0] = trialMetric;
+          if (trialMetric !== null && trialMetric > metricRange[1]) metricRange[1] = trialMetric;
```

Comment:
`nit question`

would this be easier to read as 

`metricRange[0] = Math.min(metricRange[0], trialMetric ?? Number.MAX_SAFE_INTEGER)`

Reply:
I think there are cases where the two lines would produce different results. The one I can think of is if `trialMetric` happens to be `null`. With how it's written currently, `metricRange[0]` will not actually be set to anything, where as the rewrite would produce `NaN` if `metricRange[0]` is `undefined`, or it would be set to whatever value `metricRange[0]` is set to, because it would be less than `Number.MAX_SAFE_INTEGER`.